iroh/src/commands/provide.rs

```diff
@@ -108,6 +113,45 @@ pub async fn run(
         )
     };
 
+    let db2 = db.clone();
+    let gc_task = rt.local_pool().spawn_pinned(|| async move {
+        'outer: loop {
+            tracing::info!("Starting GC mark phase");
+            let mut stream = db2.gc_mark(IrohCollectionParser, None);
+            while let Some(item) = stream.next().await {
+                match item {
+                    GcMarkEvent::CustomInfo(text) => {
+                        tracing::info!("{}", text);
+                    }
+                    GcMarkEvent::CustomWarning(text, _) => {
+                        tracing::warn!("{}", text);
+                    }
+                    GcMarkEvent::Error(err) => {
+                        tracing::error!("Fatal error during GC mark {}", err);
+                        continue 'outer;
```

Comment:
this immediately starts the GC over again, just making sure that's what we want to do on error & not sleep for 10 seconds before we try again.

Reply:
well, I was not quite sure what to do here. Exiting the loop would also be an option, but then you might lose this task and not notice. It should probably still do the delay.